### PR TITLE
add support for Serde

### LINKED ISSUE #149: Use serde instead of rustc_serialize?
Or maybe have it available a feature flag.

In theory, it doesn't REALLY matter, but in practice, I use `reqwest` for unit tests, which serializes things to JSON using serde.  So far the serde and rustc_serialize representations have always matched, but there's no guarantee that will always be the case.

I will try to work on a PR for this if you want.

### PR DESCRIPTION
**Pull request by cardoe**
Serde is the replacement for rustc-serialize so add support for it under
a feature flag so that users can continue to use rustc-serialize until
they have migrated their code. fixes #149.

### CODE REVIEW

**Comment by cardoe** [comment]
So I believe this won't break semver/ABI/API in the default case. But could definitely use some eyes and confirmations.

**Comment by tomaka** [comment]
Well, if a user has `rouille = { version = "1", default-features = true }` then this PR would remove support for rustc-serialize when they update.

**Comment by tomaka** [comment]
I think we should just remove rustc-serialize altogether.

**Comment by cardoe** [comment]
Did you mean `default-features = false`? Cause I agree that would break a user. I'm not sure how else to handle it. If I remove rustc-serialize then you'd have to go to a 2.0 since its a backwards incompatible change. You could make branch before that commit for 1.x if people are really interested in 1.x updates.

Let me know if you for sure want me to drop rustc-serialize and I'll update the PR.

**Comment by PureW** [comment]
Any update on when this could be merged?

**Comment by tomaka** [comment]
Let's just drop `rustc-serialize` altogether, make `serde` non-optional, and then bump to 2.0.

**Comment by cardoe** [comment]
Should be complete with the latest commit.

**Comment by cardoe** [comment]
Rebased to ensure it applies cleanly and still passes all tests.

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "rouille"
-version = "1.0.3"
+version = "2.0.0"
```
Comment (tomaka) [inline Cargo.toml]: Not just yet ;) Please revert, otherwise it will automatically publish 2.0.0 when this PR gets merged.

Comment (cardoe) [reply Cargo.toml]: Will do. I'll squash this in.

**Comment by cardoe** [comment]
Rebased and did the two things you requested.

**Pull request merged**